src/components/Permalink/Permalink.js

```diff
@@ -17,24 +14,119 @@ const propTypes = {
   history: PropTypes.shape({
     replace: PropTypes.func,
   }),
+
+  /**
+   * Layers provider.
+   */
+  layerService: PropTypes.instanceOf(LayerService),
+
+  /**
+   * An ol map.
+   */
+  map: PropTypes.instanceOf(OLMap),
+
+  /**
+   * Params to be written in url.
+   */
+  params: PropTypes.object,
 };
 
 const defaultProps = {
   history: null,
+  layerService: null,
+  map: null,
+  params: {},
 };
 
 /**
  * This component handles permalink logic.
  */
 class Permalink extends Component {
-  componentDidUpdate() {
+  constructor(props) {
+    super(props);
+    this.state = {
+      mapParams: null,
+      layersParams: null,
+    };
+  }
+
+  componentDidMount() {
+    const { map, layerService } = this.props;
+    if (map) {
+      map.on('moveend', () => this.onMapMoved());
+    }
+
+    if (layerService) {
+      this.updateLayerService();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { layerService } = this.props;
+
+    if (layerService !== prevProps.layerService) {
+      this.updateLayerService();
+    }
     this.updateHistory();
   }
 
+  onMapMoved() {
+    const { map } = this.props;
+    const mapView = map.getView();
+    const { center } = mapView.getProperties();
+
+    this.setState({
+      mapParams: {
```

Comment:
I don't think you need `mapParams` and `layerParams`  just put x,y,... directly in the state.